pybossa/api/task.py

```diff
@@ -72,6 +76,13 @@ def _preprocess_post_data(self, data):
         if invalid_fields:
             raise BadRequest('Missing or incorrect required fields: {}'
                             .format(','.join(invalid_fields)))
+        if data.get('gold_answers', None):
+            try:
+                gold_answers = data['gold_answers']
+                if type(gold_answers) is dict and json.dumps(gold_answers):
```

Comment:
what is the purpose of `json.dumps`?

Reply:
for json validity

Reply:
What cold go wrong? Isn't the data already parsed from json?